lib/src/socket.rs

```diff
@@ -287,9 +287,7 @@ impl SocketHandler for FrontRustls {
             }
 
             match self.session.writer().write(&buf[buffered_size..]) {
-                Ok(0) => {
-                    break;
-                }
+                Ok(0) => {} // zero byte written means that the Rustls buffers are full, we will try to write on the socket and try again
```

Comment:
do i understand that you just harmonized the behavion of the socket write accross protocols? That seems only logical

Reply:
The recommended method for writing with a `SocketHandler` is `socket_write_vectored`. This method is used by the kawa protocol in both HTTP and HTTPS sessions. However, the Pipe protocol still relies on `socket_write`, which hasn't been well-maintained, leading to this discrepancy only being noticed now. This highlights that the entire TCP and Pipe stack is somewhat outdated, and other parts related to the `socket_write` call could also use an update. This change only ensures `socket_write` has a correct behavior, similar to `socket_write_vectored.

Reply:
That makes perfect sense, thank you.